scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -819,6 +821,37 @@ def apply_filters(self, **kwargs):
                 if plat.only_tags and not set(plat.only_tags) & ts.tags:
                     instance.add_filter("Excluded tags per platform (only_tags)", Filters.PLATFORM)
 
+                if ts.required_snippets:
+                    snippet_args = {"snippets": ts.required_snippets}
+                    found_snippets = snippets.process_snippets_inter(snippet_args, [Path(ZEPHYR_BASE), Path(ts.source_dir)])
+
+                    # Search and check that all required snippet files are found
+                    for this_snippet in snippet_args['snippets']:
+                        if this_snippet not in found_snippets:
+                            instance.add_filter("Snippet not found", Filters.PLATFORM)
+                            logger.error(f"Can't find snippet '%s' for test '%s'", this_snippet, ts.name)
+
+                    # Look for required snippets and check that they are applicable for these platforms/boards
+                    for this_snippet in found_snippets:
+                        matched_snippet_board = False
+
+                        if len(found_snippets[this_snippet].appends) > 0:
+                            continue
```

Comment:
Can you explain these lines? I think I don't follow. While debugging I see that `cdc-acm-console` has `len(found_snippets[this_snippet].appends)` = 2 hence continuing, i.e. ignoring the rest of the logic, where it is decided if a matching board was found

Reply:
This checks if it is a global snippet i.e. applies to all boards, or if it only applies to some boards, see https://docs.zephyrproject.org/latest/build/snippets/writing.html#by-name for an example